tests/scalers_go/elasticsearch/elasticsearch_test.go

```diff
@@ -0,0 +1,367 @@
+            value: "false"
+          - name: ELASTIC_PASSWORD
+            value: "{{.ElasticPassword}}"
+          - name: xpack.security.enabled
+            value: "true"
+          - name: node.store.allow_mmap
+            value: "false"
+        ports:
+        - containerPort: 9200
+          name: http
+          protocol: TCP
+        - containerPort: 9300
+          name: transport
+          protocol: TCP
+        resources:
+        readinessProbe:
+          exec:
+            command:
+              - /usr/bin/curl
+              - -sS
+              - -u "elastic:{{.ElasticPassword}}"
+              - http://localhost:9200
+          failureThreshold: 3
+          initialDelaySeconds: 10
+          periodSeconds: 5
+          successThreshold: 1
+          timeoutSeconds: 5
+  serviceName: {{.DeploymentName}}
+`
+
+	serviceTemplate = `apiVersion: v1
+kind: Service
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 9200
+    targetPort: 9200
+    protocol: TCP
+  selector:
+    name: elasticsearch
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: 0
+  maxReplicaCount: 2
+  pollingInterval: 3
+  cooldownPeriod:  5
+  triggers:
+    - type: elasticsearch
+      metadata:
+        addresses: "http://{{.DeploymentName}}.{{.TestNamespace}}.svc:9200"
+        username: "elastic"
+        index: {{.IndexName}}
+        searchTemplateName: {{.SearchTemplateName}}
+        valueLocation: "hits.total.value"
+        targetValue: "1"
+        parameters: "dummy_value:1;dumb_value:oOooo"
+      authenticationRef:
+        name: keda-trigger-auth-elasticsearch-secret
+`
+
+	elasticsearchCreateIndex = `
+{
+  "mappings": {
+    "properties": {
+      "@timestamp": {
+        "type": "date"
+      },
+      "dummy": {
+        "type": "integer"
+      },
+      "dumb": {
+        "type": "keyword"
+      }
+    }
+  },
+  "settings": {
+    "number_of_replicas": 0,
+    "number_of_shards": 1
+  }
+}`
+
+	elasticsearchSearchTemplate = `
+{
+  "script": {
+    "lang": "mustache",
+    "source": {
+      "query": {
+        "bool": {
+          "filter": [
+            {
+              "range": {
+                "@timestamp": {
+                  "gte": "now-1m",
+                  "lte": "now"
+                }
+              }
+            },
+            {
+              "term": {
+                "dummy": "{{dummy_value}}"
+              }
+            },
+            {
+              "term": {
+                "dumb": "{{dumb_value}}"
+              }
+            }
+          ]
+        }
+      }
+    }
+  }
+}`
+
+	elasticsearchDummyDoc = `
+{
+  "@timestamp": "{{.Timestamp}}",
+  "dummy": 1,
+  "dumb": "oOooo"
+}`
+)
+
+var (
```

Comment:
Just a nit, could you please move this above templates, so sections of the test are aligned with the example mentioned here: https://github.com/kedacore/keda/tree/main/tests#example-test-lets-say-you-want-to-add-a-test-for-redis

Reply:
Done